backend/server.js

```diff
@@ -538,6 +529,253 @@ app.post('/subscribeEmail', wrap(async (req, res) => {
+    acc = acc || userObject.watchingSections[i].includes(req.body.classHash);
+  }
+
+  if (!acc) {
+    _.pull(userObject.watchingClasses, req.body.classHash);
+  }
+
+  const sectionInfo = req.body.sectionInfo;
+
+
+  macros.log(userObject);
+
+  await database.set(`/users/${req.body.senderId}`, userObject);
+  macros.log('sending done.');
+
+  notifyer.sendFBNotification(req.body.senderId, `You have unsubscribed from notifications for a section of ${sectionInfo.subject} ${sectionInfo.classId} (CRN: ${sectionInfo.crn}).`);
+
+  if (!acc) {
+    notifyer.sendFBNotification(req.body.senderId, `You have unsubscribed from notifications for the class ${sectionInfo.subject} ${sectionInfo.classId}`);
+  }
+
+
+  macros.log(database.get(`/user/${req.body.senderId}`));
+  // send a status of success. Hopefully it went well.
+  res.send(JSON.stringify({
+    status: 'Success',
+  }));
+}));
+
+app.post('/addClass', wrap(async (req, res) => {
+  const userObject = await sendUserHelper(req, res);
+  userObject.watchingClasses.push(req.body.info);
+
+  await database.set(`/users/${req.body.senderId}`, userObject);
+  macros.log('sending done.');
+
+  const classInfo = req.body.classInfo;
+
+
+  notifyer.sendFBNotification(req.body.senderId, `You have subscribed to notifications for the class ${classInfo.subject} ${classInfo.classId}`);
+
+  // send a status of success. Hopefully it went well.
+  res.send(JSON.stringify({
+    status: 'Success',
+  }));
+}));
+
+
+app.post('/getUserData', wrap(async (req, res) => {
+  // Don't cache this endpoint.
+  res.setHeader('Cache-Control', 'no-cache, no-store');
+
+  if (!req.body || !req.body.loginKey || (!req.body.sectionInfo && !req.body.classInfo)) {
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return;
+  }
+
+  // Checks checks checks
+  // Make sure the login key is valid
+  if (typeof req.body.loginKey !== 'string' || req.body.loginKey.length !== 100) {
+    macros.log('Invalid login key', req.body.loginKey);
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return;
+  }
+
+
+  const senderId = req.body.senderId;
+
+  // Make sure the sender id is valid
+  if (senderId && (typeof senderId !== 'string' || senderId.length !== 16 || !macros.isNumeric(senderId))) {
+    macros.log('Invalid senderId', req.body, senderId);
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return;
+  }
+
+  let matchingUser;
+
+  // If the client specified a specific senderId, lookup that specific user.
+  // if not, we have to loop over all the users's to find a matching loginKey
+
+  // long polling needs to start here
+  if (senderId) {
+    const user = await database.get(`/users/${senderId}`);
+
+    if (!user) {
+      if (reqs[req.body.loginKey]) {
+        reqs[req.body.loginKey].send(JSON.stringify({
+          error: 'Error, multiple requests from same user in quick succession',
+        }));
+      }
+      reqs[req.body.loginKey] = [res, new Date()];
```

Comment:
So here, to access res, you would have to go `reqs[req.body.loginKey][0]`, but everywhere else you access res you just go `reqs[req.body.loginKey]`, like above `reqs[req.body.loginKey].send`. 

Want to fix this up here and elsewhere? and want to make it an object {res: res, receiveTime: Date()} instead of an array?